src/storage/engine/rocksdb.rs

```diff
@@ -12,20 +12,25 @@
 // limitations under the License.
 
 use std::fmt::{self, Formatter, Debug};
+use std::sync::{Arc, Mutex};
+use threadpool::ThreadPool;
 use rocksdb::{DB, Writable, SeekKey, WriteBatch, DBIterator};
-use rocksdb::rocksdb::Snapshot as DBSnapshot;
 use kvproto::kvrpcpb::Context;
 use storage::{Key, Value, CfName};
+use raftstore::store::engine::{Snapshot as RocksSnapshot, Peekable, Iterable};
 use util::escape;
 use util::rocksdb;
-use super::{Engine, Snapshot, Modify, Cursor, TEMP_DIR, Result, Error, DEFAULT_CFNAME};
+use super::{Engine, Snapshot, Modify, Cursor, Callback, TEMP_DIR, Result, Error, DEFAULT_CFNAME};
 use tempdir::TempDir;
 
 
 pub struct EngineRocksdb {
-    db: DB,
+    db: Arc<DB>,
     // only use for memory mode
     temp_dir: Option<TempDir>,
+    // ThreadPool is handy for executing task asynchronously, each Engine will
+    // contain only 1 thread.
+    pool: Mutex<ThreadPool>,
```

Comment:
To use `util::worker`, we typically will have to define a enum Msg, then implement a `Runner` to handle different types of Msgs.
While `ThreadPool` has a `fn execute<F>(&self, job: F) where F: FnOnce() + Send + 'static`, which is more easy to use.